src/utils.ts

```diff
@@ -173,6 +195,37 @@ export async function resolveLabel(label: string, config: Config): Promise<strin
   return config.provider.resolveName(`${label}.${config.registrar.domain}`);
 }
 
+export async function lookupAddress(address: string, config: Config): Promise<Profile | null>  {
+  const profile: Profile = {} as Profile;
+  try {
+
+    const ens = await resolveEnsProfile(address, config);
+    if (ens) profile.ens = ens;
+    const idx = await resolveIdxProfile(formatCAIP10Address(address, "eip155", 3), config);
```

Comment:
We don't want to execute these serially, as they don't depend on each other. You can use `Promise.allSettled`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled